Source/WebGPU/WebGPU/RenderPassEncoder.mm

```diff
@@ -138,6 +138,22 @@
 
         addResourceToActiveResources(textureView, (isReadOnly && *isReadOnly) ? BindGroupEntryUsage::AttachmentRead : BindGroupEntryUsage::Attachment);
     }
+
+    m_viewportWidth = static_cast<float>(m_renderTargetWidth);
+    m_viewportHeight = static_cast<float>(m_renderTargetHeight);
+}
+
+double RenderPassEncoder::quantizedDepthValue(double depthClearValue, WGPUTextureFormat pixelFormat)
+{
+    if (depthClearValue < 0 || depthClearValue > 1)
+        return depthClearValue;
+
+    switch (pixelFormat) {
+    case WGPUTextureFormat_Depth16Unorm:
+        return depthClearValue + 1.0 / USHRT_MAX;
```

Comment:
This might be a Metal driver bug, I need to investigate more closely. But I think more accurately it is because 0.5 can not be representing as a unorm with 16-bit precision

Reply:
What's the issue? (Yes, 0.5 isn't exactly representable)

Reply:
if I write 0.5 to clearValue, then it comes across as 0.499992 for Depth16Unorm depth target.

If 0.5 is written out from the fragment shader to the depth value, and depth comparison function of `MTLCompareFunctionNotEqual` is used, then it is determined that 0.5 != 0.499992 (the value in the depth target) so the write occurs when it should not.

Adding 1.0 / uint16_max for the clear value gets a value of '0.5' in the tools debugger for Depth16Unorm and then using `MTLCompareFunctionNotEqual` later does not commit the color write when the depth is written out as 0.5.

It really seems like a driver bug or at least an inconsistency. Setting 0.5 as the depth clear value and writing 0.5 out from the fragment shader to the depth should result in the same value.

Reply:
I also tried 0.5 / uint16_max thinking something was rounding up when it should not. But the depth clear value was still too low (less than 0.5).

If I can make a repro case I will report a bug

Reply:
@djg so this whole extra pass was completely unnecessary. I added it before I added the WGSL code generation, thinking we could do without extra instructions in the generated WGSL code. However, the CTS revealed it was inadequate and the code-gen was needed.

With the extra codegen, we don't need the extra render pass, so I just removed it.